protected/humhub/modules/content/widgets/stream/StreamEntryWidget.php

```diff
@@ -102,15 +102,31 @@ private static function renderLegacyWallEntry(ContentActiveRecord $record, $opti
             $options = [];
         }
 
+        // Ensure that $record->getWallEntryWidget() returns a valid object
+        $wallEntryWidget = $record->getWallEntryWidget();
```

Comment:
I think the `$wallEntryWidget` should be defined only for case below when `!isset($options['jsWidget'])`.

Reply:
I mean your code now is:

```php
        $wallEntryWidget = $record->getWallEntryWidget();
        ...
        if (isset($options['jsWidget'])) {
            $jsWidget = $options['jsWidget'];
            unset($options['jsWidget']);
        } else {
            $jsWidget = $wallEntryWidget->renderOptions->jsWidget;
        }
```

So why should we define the `$wallEntryWidget` even for case `if (isset($options['jsWidget'])) {` when the var is not used.
Why don't move it right after the line `} else {`?

Reply:
> I mean your code now is:
> 
> ```
>         $wallEntryWidget = $record->getWallEntryWidget();
>         ...
>         if (isset($options['jsWidget'])) {
>             $jsWidget = $options['jsWidget'];
>             unset($options['jsWidget']);
>         } else {
>             $jsWidget = $wallEntryWidget->renderOptions->jsWidget;
>         }
> ```
> 
> So why should we define the `$wallEntryWidget` even for case `if (isset($options['jsWidget'])) {` when the var is not used. Why don't move it right after the line `} else {`?

This way, `$wallEntryWidget` is always defined, and you won't encounter the "Undefined variable" error when using the search for content.

Reply:
If it's made conditional the error will still happen, I've tested this and the error pops up when using the "Show all results" for content.